adoc-mode.el

```diff
@@ -46,6 +46,10 @@
 (require 'tempo)
 (require 'mouse)
 (require 'image)
+(eval-when-compile
+  (when (version< emacs-version "29.0") ;; when-let has moved from subr-x to subr
+    (require 'subr-x)
```

Comment:
subr-x is full of useful functions, so it's fine to just require it unconditionally.